src/video/x11/SDL_x11events.c

```diff
@@ -1377,6 +1377,27 @@ X11_DispatchEvent(_THIS)
                     printf("New _NET_FRAME_EXTENTS: left=%d right=%d, top=%d, bottom=%d\n", data->border_left, data->border_right, data->border_top, data->border_bottom);
                     #endif
                 }
+            } else {
+                char* name_of_atom = X11_XGetAtomName(display, xevent.xproperty.atom);
+
+                if (SDL_strncmp(name_of_atom, "_ICC_PROFILE", sizeof("_ICC_PROFILE")) == 0) {
+                    XWindowAttributes attrib;
+                    int screennum;
+
+                    X11_XGetWindowAttributes(display, data->xwindow, &attrib);
+                    screennum = X11_XScreenNumberOfScreen(attrib.screen);
+                    if (screennum == 0 && SDL_strcmp(name_of_atom, "_ICC_PROFILE") == 0) {
+                        SDL_SendWindowEvent(data->window, SDL_WINDOWEVENT_ICCPROF_CHANGED, 0, 0);
+                    } else if (SDL_strncmp(name_of_atom, "_ICC_PROFILE_", sizeof("_ICC_PROFILE_")) == 0 && SDL_strlen(name_of_atom) > sizeof("_ICC_PROFILE_") - 1) {
+                        int iccscreennum = SDL_atoi(&name_of_atom[sizeof("_ICC_PROFILE_")]);
```

Comment:
This should also be `sizeof("_ICC_PROFILE") - 1`, otherwise it points to the `\0` byte.

Reply:
Is that needed? Because this will directly point to the number string as the `strlen` check ensures that the atom name is larger than `_ICC_PROFILE_` according to my understanding.

Reply:
The problem is that `sizeof("_ICC_PROFILE_")` includes the size of the terminating `\0` byte, so in this case, it's **14**, not 13 as you might expect.

Indexing `&name_of_atom[14]`, therefore, points to the byte *after* the one you're interested in, which will generally be the trailing `\0` terminator. You can check this easily yourself:

```
#include <stdio.h>
#include <string.h>

int main() {
    const char *name_of_atom = "_ICC_PROFILE_2";

    printf("sizeof: <%s>\n", &name_of_atom[sizeof("_ICC_PROFILE_")]); // prints empty string
    printf("strlen: <%s>\n", &name_of_atom[strlen("_ICC_PROFILE_")]); // prints `2`
}
```

Reply:
Btw, the code would probably be cleaner if you extract the number from the atom first and then iterate over the windows, rather than doing it in every loop iteration.

Reply:
That would result in issues later because the ICC profile atoms are sort of dynamic, unlike other atoms SDL currently uses.